File: sdk/java/src/main/java/com/lightswitch/domain/dto/BaseResponse.java
package com.lightswitch.domain.dto;

public class BaseResponse<T> {

	private int code;
	private String message;
	private T data;

	public BaseResponse(int code, String message, T data) {
		this.code = code;
		this.message = message;
		this.data = data;
	}

	public int getCode() {
		return code;
	}

	public T getData() {
		return data;
	}

	public void setCode(int code) {
		this.code = code;
	}

	public void setData(T data) {
		this.data = data;
	}

	public void setMessage(String message) {
		this.message = message;
	}
}

File: backend/core-service/src/main/kotlin/com/lightswitch/core/common/dto/BaseResponse.kt
package com.lightswitch.core.common.dto

class BaseResponse<T>(
    val code: Int,
    val message: String,
    val data: T
)

fun <T> success(data: T): BaseResponse<T> {
    return BaseResponse(ResponseCode.OK.code, ResponseCode.OK.message, data)
}

fun <T> fail(responseCode: ResponseCode, data: T): BaseResponse<T> {
    return BaseResponse(responseCode.code, responseCode.message, data)
}